Assets/UnityPharusAPI/Managers/ATracklinkPlayerManager.cs

```diff
@@ -94,8 +94,11 @@ public virtual void AddPlayer(TrackRecord trackRecord)
             aPlayer.Orientation = new Vector2(trackRecord.orientation.x, trackRecord.orientation.y);
             aPlayer.Speed = trackRecord.speed;
             aPlayer.Echoes.Clear();
-            trackRecord.echoes.AddToVector2List(VectorAdapter.ToPharusVector2List(aPlayer.Echoes));
-
+            // trackRecord.echoes.AddToVector2List(VectorAdapter.ToPharusVector2List(aPlayer.Echoes));
+            foreach (KeyValuePair<int, Vector2> keyValuePair in aPlayer.Echoes)
+            {
+                trackRecord.echoes.Add(new Vector2f(keyValuePair.Value.x, keyValuePair.Value.y));
+            }
```

Comment:
My C# is a bit rusty but I sense a misunderstanding here: 

The original code was actually adding `trackRecord.echoes` to `aPlayer.Echoes`.  
```
trackRecord.echoes.AddToVector2List(VectorAdapter.ToPharusVector2List(aPlayer.Echoes));
``` 
Please have a look at the extension method to see what I mean: https://github.com/Playful-Interactive-Environments/UnityTrackingClient/blob/5d36379da3bf9ce9cdf56679c84d576dc71ed312/UnityPharusAPI/UnityPharusAPI/TransmissionFrameworks/Tracklink/UnityPharusUtils.cs#L22

This new loop however will do the exact oposite (adding `aPlayer.Echoes` to `trackRecord.echoes`) which is probably not intended. 
```
foreach (KeyValuePair<int, Vector2> keyValuePair in aPlayer.Echoes)
{
    trackRecord.echoes.Add(new Vector2f(keyValuePair.Value.x, keyValuePair.Value.y));
}
```
I did not test this but I am pretty certain that as a result `aPlayer.Echoes` will remain empty (as it was cleared in line 96).

Reply:
Yep, definitely correct. I remember working on this while I was writing my master thesis so I needed to look a little bit more to understand what I actually had to do. My new commit should resolve all the problems!